components/decimal/src/lib.rs

```diff
@@ -170,3 +166,35 @@ impl FixedDecimalFormatter {
         self.format(value).write_to_string().into_owned()
     }
 }
+
+icu_preferences::preferences!(
+    /// TODO
+    FixedDecimalFormatterPreferences,
+    ResolvedFixedDecimalFormatterPreferences,
+    {
+        /// TODO
+        numbering_system: NumberingSystem
+    }
+);
+
+impl FixedDecimalFormatterPreferences<'_> {
+    /// TODO
+    pub fn resolve(self) -> ResolvedFixedDecimalFormatterPreferences {
+        #[allow(clippy::unwrap_used)] // temporary
+        ResolvedFixedDecimalFormatterPreferences {
+            data_locale: DataLocale::from(self.lid),
+            numbering_system: self.numbering_system.unwrap(),
+        }
+    }
+}
+
+impl ResolvedFixedDecimalFormatterPreferences {
+    fn as_data_id(&self) -> DataIdentifierBorrowed {
+        DataIdentifierBorrowed::for_marker_attributes_and_locale(
+            DataMarkerAttributes::from_str_or_panic(
+                self.numbering_system.0.as_str()
+            ),
+            &self.data_locale,
+        )
+    }
+}
```

Comment:
Praise: I like the pattern of pre-computing the DataLocale and putting this code into an associated function.